packages/react-native-renderer/src/ReactNativeGetListeners.js

```diff
@@ -0,0 +1,168 @@
+  if (props === null) {
+    // Work in progress.
+    return null;
+  }
+
+  const listener = props[registrationName];
+
+  if (listener && typeof listener !== 'function') {
+    throw new Error(
+      `Expected \`${registrationName}\` listener to be a function, instead got a value of \`${typeof listener}\` type.`,
+    );
+  }
+
+  // If there are no imperative listeners, early exit.
+  if (
+    !(
+      dispatchToImperativeListeners &&
+      stateNode.canonical &&
+      stateNode.canonical._eventListeners
+    )
+  ) {
+    return listener;
+  }
+
+  // Below this is the de-optimized path.
+  // If you are using _eventListeners, we do not (yet)
+  // expect this to be as performant as the props-only path.
+  // If/when this becomes a bottleneck, it can be refactored
+  // to avoid unnecessary closures and array allocations.
+  //
+  // Previously, there was only one possible listener for an event:
+  // the onEventName property in props.
+  // Now, it is also possible to have N listeners
+  // for a specific event on a node. Thus, we accumulate all of the listeners,
+  // including the props listener, and return a function that calls them all in
+  // order, starting with the handler prop and then the listeners in order.
+  // We return either a non-empty array or null.
+  const listeners = [];
+  if (listener) {
+    listeners.push(listener);
+  }
+
+  // TODO: for now, all of these events get an `rn:` prefix to enforce
+  // that the user knows they're only getting non-W3C-compliant events
+  // through this imperative event API.
+  // Events might not necessarily be noncompliant, but we currently have
+  // no verification that /any/ events are compliant.
+  // Thus, we prefix to ensure no collision with W3C event names.
+  const requestedPhaseIsCapture = phase === 'captured';
+  const mangledImperativeRegistrationName = requestedPhaseIsCapture
+    ? 'rn:' + registrationName.replace(/Capture$/, '')
+    : 'rn:' + registrationName;
+
+  // Get imperative event listeners for this event
+  if (
+    stateNode.canonical._eventListeners[mangledImperativeRegistrationName] &&
+    stateNode.canonical._eventListeners[mangledImperativeRegistrationName]
+      .length > 0
+  ) {
+    const eventListeners =
+      stateNode.canonical._eventListeners[mangledImperativeRegistrationName];
+
+    eventListeners.forEach(listenerObj => {
+      // Make sure phase of listener matches requested phase
+      const isCaptureEvent =
+        listenerObj.options.capture != null && listenerObj.options.capture;
+      if (isCaptureEvent !== requestedPhaseIsCapture) {
+        return;
+      }
+
+      // For now (this is an area of future optimization) we must wrap
+      // all imperative event listeners in a function to unwrap the SyntheticEvent
+      // and pass them an Event.
+      // When this API is more stable and used more frequently, we can revisit.
+      const listenerFnWrapper = function(syntheticEvent) {
```

Comment:
How come we don't grab the args on this listenerFnWrapper to pass on line 130, like we do in the case of line 138?

Reply:
In theory this is the only argument and we need to do something with specificEvent... in the code below, I don't really need to do anything with the args so I just pass them on without assuming anything. Practically it could be `event` instead of `...args` below, but theoretically `...args` is more flexible? Doesn't really matter, just personal preference I guess.

Reply:
@lunaleaps how's the current impl?

I don't want to pull out event where it's not actually used, but it makes sense to always have rest/spread args.